### PR TITLE
Recoverable Transaction History Bindings for iOS

### PR DESCRIPTION
**Pull request by the-real-adammork**
Soundtrack of this PR: [Scruscru - Chewing Gum (feat. Sunner Soul)](https://youtu.be/79uwXIcuqvI)

### Motivation

iOS Bindings for Recoverable Transaction History (v1.2.0)

### In this PR
* iOS FFI/Bindings needed for the SDK to implement Recoverable Transaction History

### TODO Before Merge
* Update `rust-bitcode` script to use the new rust nightly that master is expecting.
* merge in `master` after this PR merges https://github.com/mobilecoinfoundation/mobilecoin/pull/1606

### CODE REVIEW

File: libmobilecoin/libmobilecoin_cbindgen.h [hunk 1]
```diff
@@ -769,6 +777,27 @@ FfiOptOwnedPtr<McData> mc_transaction_builder_add_output(FfiMutPtr<McTransaction
                                                          FfiMutPtr<McMutableBuffer> out_tx_out_confirmation_number,
                                                          FfiOptMutPtr<FfiOptOwnedPtr<McError>> out_error);
 
+/**
+ * # Preconditions
+ *
+ * * `account_kay` - must be a valid account key, default change address
```
Comment (garbageslam) [inline libmobilecoin/libmobilecoin_cbindgen.h:783]:
```suggestion
 * * `account_key` - must be a valid account key, default change address
```

**Review by garbageslam — APPROVED** [review]
LGTM

**Review by jcape — APPROVED** [review]
LGTM

**Comment by jcape** [comment]
This PR is failing to build because of out-of-date Cargo.lock files. You should run a build and check in the changed Cargo.lock versions---please don't merge until someone has reviewed the changes to Cargo.lock, though :-)

**Comment by the-real-adammork** [comment]
@jcape updated with the new cargo lock and linting/formatting changes

File: transaction/std/Cargo.toml [hunk 2]
```diff
@@ -11,6 +11,7 @@ test-only = []
 # External dependencies
 cfg-if = "1.0"
 displaydoc = { version = "0.2", default-features = false }
+hex = "0.4.3"
```
Comment (garbageslam) [inline transaction/std/Cargo.toml:14]: Is this actually needed?

Comment (the-real-adammork) [reply transaction/std/Cargo.toml]: ya its not, it was part of the android-rth changes but those aren't in this pr. removed.

**Pull request merged**